submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/pagetab/FirePageTabService.kt

```diff
@@ -21,53 +20,54 @@ class FirePageTabService(
         val subFiles = serializationService.generateSubPageTab(sub, fireTempFolder)
         val fileListFiles = serializationService.generateFileListPageTab(sub, fireTempFolder)
 
-        val section = process(sub.section) { updateFileList(it, sub.filesPath, fileListFiles) }
+        val section = process(sub.section) { updateFileList(it, sub.relPath, fileListFiles) }
 
         return when {
-            section.changed -> sub.copy(pageTabFiles = extFiles(sub.filesPath, subFiles), section = section.section)
-            else -> sub.copy(pageTabFiles = extFiles(sub.filesPath, subFiles))
+            section.changed -> sub.copy(pageTabFiles = subExtFiles(subFiles, sub.relPath), section = section.section)
+            else -> sub.copy(pageTabFiles = subExtFiles(subFiles, sub.relPath))
         }
     }
 
     private fun updateFileList(sec: ExtSection, path: String, tab: Map<String, TabFiles>): Section {
         return when (val lst = sec.fileList) {
             null -> Section(false, sec)
-            else -> Section(
-                true, sec.copy(fileList = lst.copy(pageTabFiles = extFiles(path, tab.getValue(lst.fileName))))
-            )
+            else -> {
+                val name = lst.fileName
+                val tabFiles = tab.getValue(name)
+                Section(true, sec.copy(fileList = lst.copy(pageTabFiles = fileListFiles(tabFiles, path, name))))
+            }
         }
     }
 
-    private fun extFiles(filesRelPath: String, pageTab: TabFiles): List<ExtFile> {
-        val json = fireWebClient.save(pageTab.json, pageTab.json.md5(), filesRelPath)
-        val xml = fireWebClient.save(pageTab.xml, pageTab.xml.md5(), filesRelPath)
-        val tsv = fireWebClient.save(pageTab.tsv, pageTab.tsv.md5(), filesRelPath)
+    private fun fileListFiles(pageTab: TabFiles, subFolder: String, fileListName: String) = listOf(
+        saveFileListFile(pageTab.json, subFolder, "$fileListName.json"),
+        saveFileListFile(pageTab.xml, subFolder, "$fileListName.xml"),
+        saveFileListFile(pageTab.tsv, subFolder, "$fileListName.pagetab.tsv")
+    )
 
-        val extJson = FireFile(
-            pageTab.json.name,
-            pageTab.json.path,
-            json.fireOid,
-            json.objectMd5,
-            json.objectSize.toLong(),
-            listOf()
-        )
-        val extXml = FireFile(
-            pageTab.xml.name,
-            pageTab.xml.path,
-            xml.fireOid,
-            xml.objectMd5,
-            xml.objectSize.toLong(),
-            listOf()
-        )
-        val extTsv = FireFile(
-            pageTab.tsv.name,
-            pageTab.tsv.path,
-            tsv.fireOid,
-            tsv.objectMd5,
-            tsv.objectSize.toLong(),
-            listOf()
+    private fun saveFileListFile(file: File, subFolder: String, filePath: String): FireFile {
+        val relPath = "Files/$filePath"
+        val db = fireWebClient.save(file, file.md5(), "$subFolder/$relPath")
+        return FireFile(
```

Comment:
If we don't use named params, could this fit in a single line like [here](https://github.com/EBIBioStudies/biostudies-backend-services/pull/456/files#diff-6de59b48d495dece61110c80d0ead258363f3dfe275132ace53f5d49d464a903R66)?